pkg/compose/exec.go

```diff
@@ -38,7 +39,7 @@ func (s *composeService) Exec(ctx context.Context, project string, opts api.RunO
 
 	exec, err := s.apiClient.ContainerExecCreate(ctx, container.ID, moby.ExecConfig{
 		Cmd:        opts.Command,
-		Env:        opts.Environment,
+		Env:        ToMobyEnv(types.NewMappingWithEquals(opts.Environment).Resolve(os.LookupEnv)),
```

Comment:
Resolve should use project.Environment, not os.LookupEnv, so this also can resolve variables set by .env, and let us (in the future) get more control on the way environment is set

Reply:
Hmmm... I think we need to change the interface for that. To construct the whole project we need to do that in the CLI side. So before arriving here.